.jenkins/Jenkinsfile

```diff
@@ -250,227 +198,149 @@ def isCanonical() {
+        cleanWs(disableDeferredWipeout: true)
+        fetchSource(cell.step, cell.arch, cell.jdk)
+        def logfile = "${JOB_NAME}_${BUILD_NUMBER}_${cell.step}_jdk${cell.jdk}_python_${cell.python}_${cell.cython}_${cell.arch}.log"
+
+        def cassandra_dtest_dir = "${WORKSPACE}/build/cassandra-dtest"
+        if (command.containsKey('dtest') && command['dtest']) {
+          checkout changelog: false, poll: false, scm: scmGit(branches: [[name: 'trunk']], extensions: [cloneOption(depth: 1, noTags: true, reference: '', shallow: true), [$class: 'RelativeTargetDirectory', relativeTargetDir: "${WORKSPACE}/build/cassandra-dtest"]], userRemoteConfigs: [[url: 'https://github.com/apache/cassandra-dtest']])
         }
-        echo "Executing ${STAGE_NAME} `build_dir=${build_dir} ${build_script} ${jdk} 2>&1 | tee ${logfile}` on ${NODE_NAME}"
-        statusCode = sh returnStatus:true, script:"build_dir=\"${build_dir}\" ${build_script} ${jdk} 2>&1 | tee ${build_dir}/${logfile}"
 
-        if ("jar" == STAGE_NAME) {
-          dir("${build_dir}") {
-            // only stash the project built files. all dependency libraries are restored from the local maven repo using `ant resolver-dist-lib`
-            stash name: "${arch}_${jdk}"//, includes: "*.jar,classes/**,test/classes/**,tools/**"
+  // sh "mkdir -p build"
+  // def build_dir = sh( returnStdout: true, script:"mktemp -d ${WORKSPACE}/build/build_docker_run.XXXXXXXXXX" ).trim()
+  // def build_dir_rel = build_dir - "${WORKSPACE}/"
+
+  // dir("${build_dir}") {
+    // fetchSource(cell.step, arch, jdk)
+  // }
+        def stageError = true
+        catchError(buildResult: null, message: 'Tests failed', stageResult: 'FAILURE') {
+          sh label: "RUNNING TESTS...", script: "#!/usr/bin/bash \nset -o pipefail;  DEBUG=true python_version='${cell.python}' cython='${cell.cython}' build_dir='${WORKSPACE}/build' cassandra_dtest_dir='${cassandra_dtest_dir}' .build/docker/run-tests.sh ${cell.step} '${cell.split}/${splits}' ${cell.jdk} 2>&1 | tee ${logfile}"
+          dir("build") {
+            // junit testResults: "test/output/nosetests.xml", allowEmptyResults: true
+            // junit testResults: "test/output/cqlshlib.xml", allowEmptyResults: true
+            //sh "ls -la test/output/"
+            junit testResults: "test/output/**/TEST*.xml", allowEmptyResults: true
+            sh "xz -f test/output/**/TEST*.xml || true ; xz -f test/output/cqlshlib.xml test/output/nosetests.xml || true"
+            // junit testResults: "test/output/**/TEST*.xml,test/output/cqlshlib.xml,test/output/nosetests.xml", testDataPublishers: [[$class: 'StabilityTestDataPublisher']]
+            // junit testResults: "build/test/output/**/TEST*.xml,build/test/output/cqlshlib.xml,build/test/output/nosetests.xml"//, testDataPublishers: [[$class: 'StabilityTestDataPublisher']]
+            // sh "xz -f build/test/output/**/TEST*.xml || true ; xz -f build/test/output/cqlshlib.xml build/test/output/nosetests.xml || true"
```

Comment:
needs cleanup.

remove ` allowEmptyResults: true`
needs `testDataPublishers: [[$class: 'StabilityTestDataPublisher']]`

Reply:
done in my commit